File: test/fib.lua
---SPDX-FileCopyrightText: Michael Fisher @mfisher31
---SPDX-License-Identifier: MIT

local function fib(n)
    if n < 2 then
        return n;
    end
    return fib(n - 2) + fib(n - 1);
end

fib(40)

File: test/fib.cpp
// SPDX-FileCopyrightText: Michael Fisher @mfisher31
// SPDX-License-Identifier: MIT

static double fib (double x) {
    if (x < 2.0)
        return x;
    return fib (x - 2.0) + fib (x - 1.0);
}

int main() {
    fib (40.0);
    return 0;
}
